onadata/apps/api/viewsets/floip_viewset.py

```diff
@@ -74,6 +117,26 @@ def get_object(self):
             self.permission_denied(self.request)
         return obj
 
+    def filter_queryset(self, queryset):
+        if self.action == 'responses' and queryset.model == Instance:
+            for fil_key, fil_value in self.request.query_params.items():
+                if fil_key in self.filter_map:
+                    if fil_key in [
+                            "filter[max-version]", "filter[min-version]"]:
+                        fil_value = dateutil.parser.parse(fil_value)
+                        fil_value = fil_value.strftime("%Y%m%d%H%M")
+
+                    if fil_key in ["page[afterCursor]", "page[beforeCursor]"]:
+                        fil_value, _ = inverse_pairing(int(fil_value))
```

Comment:
Retrieves the Instance ID from the `row_id` value which is usually a combination of two numbers. _Retrieving it so we can decide which instance to start from to make it easier to yield the responses from the selected cursor in later steps_